vertx-web/src/main/java/io/vertx/ext/web/impl/RoutingContextImpl.java

```diff
@@ -364,6 +366,38 @@ public void reroute(HttpMethod method, String path) {
     return getPathParams().get(name);
   }
 
+  @Override
+  public Map<String, String> queryParams() {
+    return getQueryParams();
+  }
+
+  @Override
+  public @Nullable List<String> queryParam(String query) {
+    return getQueryParamsParsed().get(query);
+  }
+
+  protected void setQueryParams(Map<String, List<String>> q) {
+    queryParamsParsed = new HashMap<>();
+    queryParamsParsed.putAll(q);
+    queryParams = new HashMap<>();
+    for (Map.Entry<String, List<String>> e : q.entrySet())
+      queryParams.put(e.getKey(), String.join(",", e.getValue()));
+  }
+
+  private Map<String, String> getQueryParams() {
+    if (queryParams == null) {
+      queryParams = new HashMap<>();
```

Comment:
The real trouble is that `HttpServerRequest.params()` don't return only query params, It returns an unique multimap with path params and query params. I've done a small test:
    Router router = Router.router(vertx);

        router.route().handler(BodyHandler.create());
        router.route("/:id").handler(routingContext -> {
            System.out.println(Arrays.toString(routingContext.request().params().entries().toArray()));
           routingContext.response().end("Hello!");
        });

        vertx.createHttpServer().requestHandler(router::accept).listen(8080);

Doing a request with this URL:
    http://localhost:8080/ciao?ei=ao

It prints in console this:
    [ei: ao, id: ciao]